orderbook/src/api/get_fee_and_quote.rs

```diff
@@ -8,70 +8,104 @@ use serde::{Deserialize, Serialize};
 use shared::price_estimation::{self, PriceEstimating, PriceEstimationError};
 use std::convert::Infallible;
 use std::sync::Arc;
+use warp::reply::Json;
 use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
 
 #[derive(Serialize)]
 #[serde(rename_all = "camelCase")]
-struct Fee {
+pub struct Fee {
```

Comment:
I'm not a huge fan of this. Instead of making this fee route public, why not make the logic in `post_quote` public and turn these `{Sell,Buy}Query` into a `QuoteOrderQuery` and then take the resulting `OrderQuote` and convert it into a `{Sell,Buy}Response`.

Reply:
Yes I would very much like to not make these things public. I will try and do so although I am not aware of any {Sell, Buy}Responses that exist already for which this would prevent making Fee stuff public.

Reply:
My point is that instead of `post_quote` API route using the `get_fee_and_quote` logic, we should move the logic into `post_quote` and make `get_fee_and_quote` use that instead.

Reply:
Ok, but then wont I just have to make it public in a different place?

Reply:
> I  just have to make it public in a different place?

Yes, this is what I meant with "why not make the logic in `post_quote` public".

Reply:
Excellent, this is a very good idea and I am in the process of making it a reality. However, I worry that this is going to become a much bigger change than originally expected. May have to split it up...